src/d3a/models/market/two_sided.py

```diff
@@ -349,15 +349,20 @@ def validate_bid_offer_match(
         # All combinations of bids and offers [(bid, offer), (bid, offer)...]
         # Example List1: [A, B], List2: [C, D] -> combinations: [(A, C), (A, D), (B, C), (B, D)]
         bids_offers_combinations = itertools.product(bids, offers)
-        if not (
-                bids_total_energy >= selected_energy and
-                offers_total_energy >= selected_energy
-                and all(
-                    (bid.energy_rate + FLOATING_POINT_TOLERANCE) >= clearing_rate for bid in bids)
-                and all(
-                    (offer.energy_rate <= clearing_rate + FLOATING_POINT_TOLERANCE
-                        for offer in offers))):
-            raise InvalidBidOfferPairException
+        if selected_energy > bids_total_energy:
+            raise InvalidBidOfferPairException(
+                f"Energy traded {selected_energy} is higher than bids energy {bids_total_energy}.")
+        if selected_energy > offers_total_energy:
+            raise InvalidBidOfferPairException(
+                f"Energy traded {selected_energy} is higher than offers energy"
+                f" {offers_total_energy}.")
+        if any((bid.energy_rate + FLOATING_POINT_TOLERANCE) < clearing_rate for bid in bids):
```

Comment:
If we really want the requirements matching to act as pay as bid, it should be this way IMO:

```suggestion
        if any(not isclose(bid.energy_rate + FLOATING_POINT_TOLERANCE, clearing_rate) for bid in bids):
```

Reply:
In my understanding the match can be valid if the clearing rate is less than or equal to the bid's rate and greater than or equal to the offer's rate

Reply:
@candrews42

Reply:
Since this is a generic validator, and not trying to match the 2-sided PAB market, then I agree, you can leave it as is.